### DISCUSSION TITLE
Support for aws data api

### DISCUSSION THREAD

**Discussion by alexstolr** [sqlalchemy/sqlalchemy#8818] [Usage Questions]
"The Data API doesn't require a persistent connection to the DB cluster. Instead, it provides a secure HTTP endpoint and integration with AWS SDKs" - https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/data-api.html

I coud'nt find anywhere in the documentation a way to use SqlAlchemy without creating a connection. Is there a way to use SqlAlchemy with aws serverless data api -  without maintaining a connection?

I could find [this](https://github.com/koxudaxi/py-data-api) implementation but it seems unmaintained. 

Any help is appreciated.

**Comment by CaselIT** [comment]
Hi,

I think this can be achieved by using a null pool, that will create a new connection every time one is required.
As for dialects, I don't know if there are any for that service.

**Reply by alexstolr** [reply]
data api is connectionless, it sends http requests instead of having a tcp connection, so a connection isn't required at all. Perhaps I didn't understand you'r answer, could you elaborate please?

**Reply by CaselIT** [reply]
oh ok.
Then all this would be managed by the driver that sqlalchemy will use.

If you question is can you use sqlalchemy without an sqlalchemy connection, the answer is no. But a sqlalchemy connection is just an abstraction, the driver can do what it prefers. I'm guessing all this is managed by the driver you linked
